libraries/package-deps-hash/src/getRepoState.ts

```diff
@@ -246,6 +246,109 @@ export function applyWorkingTreeState(
   }
 }
 
+/**
+ * Get the list of all git submodule paths
+ * @param rootDirectory - The root directory of the Git repository
+ * @param gitPath - The path to the Git executable
+ * @internal
+ */
+export function getSubmodulePaths(rootDirectory: string, gitPath?: string): string[] {
+  const submodulePaths: string[] = [];
+
+  const submoduleStatusResult: child_process.SpawnSyncReturns<string> = Executable.spawnSync(
+    gitPath || 'git',
+    ['submodule', 'status'],
+    {
+      currentWorkingDirectory: rootDirectory
+    }
+  );
+
+  if (submoduleStatusResult.status !== 0) {
+    ensureGitMinimumVersion(gitPath);
+
+    throw new Error(
+      `git submodule status exited with status ${submoduleStatusResult.status}: ${submoduleStatusResult.stderr}`
+    );
+  }
+
+  // <hash> <path> (heads/main)\n
+  submoduleStatusResult.stdout.split('\n').forEach((line) => {
+    if (line) {
+      const [, submodulePath] = line.trim().split(' ');
+      submodulePaths.push(submodulePath);
+    }
+  });
+  return submodulePaths;
+}
+
+/**
+ * Augments the state value with modifications that are in git submodules
+ * @param rootDirectory - The root directory of the Git repository
+ * @param state - The current map of git path -> object hash. Will be mutated.
+ * @param gitPath - The path to the Git executable
+ * @internal
+ */
+export function applySubmodulesTreeState(
+  rootDirectory: string,
+  state: Map<string, string>,
+  gitPath?: string
+): void {
+  const submodulePaths: string[] = getSubmodulePaths(rootDirectory, gitPath);
+
+  if (submodulePaths.length === 0) {
+    return;
+  }
+
+  const submoduleLsTreeResult: child_process.SpawnSyncReturns<string> = Executable.spawnSync(
+    gitPath || 'git',
+    ['submodule', 'foreach', 'git', '--no-optional-locks', 'ls-tree', '-r', '--full-name', 'HEAD', '--'],
```

Comment:
Could you test your PR with an oldest reasonable version of Git, and also a very recent version of Git, to ensure that all these parameters are allowed and the output is as expected?

Reply:
Tested it works in the following platform ✅
- git@2.37.1 on MacOS 
- git@2.20.1 on Linux